runtime/js-compute-runtime/builtins/shared/text-decoder.cpp

```diff
@@ -198,18 +199,17 @@ bool TextDecoder::constructor(JSContext *cx, unsigned argc, JS::Value *vp) {
   // 1. Remove any leading and trailing ASCII whitespace from label.
   // 2. If label is an ASCII case-insensitive match for any of the labels listed in the table
   // below, then return the corresponding encoding; otherwise return failure. JS-Compute-Runtime:
-  size_t length;
   jsencoding::Encoding *encoding;
   if (label_value.isUndefined()) {
     encoding = const_cast<jsencoding::Encoding *>(jsencoding::encoding_for_label_no_replacement(
         reinterpret_cast<uint8_t *>(const_cast<char *>("UTF-8")), 5));
   } else {
-    auto label_chars = encode(cx, label_value, &length);
+    auto label_chars = fastly::core::encode(cx, label_value);
```

Comment:
Will this `fastly::core` automatically be pulled into the shared builtin libraries? If so can we work towards making this a vendor neutral approach for ComponentizeJS?

Reply:
We could definitely do that. I'm not attached to using the top-level `fastly` namespace, but I would like to start using namespaces a bit more aggressively in the codebase to avoid name clashes that come up periodically.

Reply:
If we can continue to treat the builtins folder as an abstraction that can move towards being runtime netural I think that would be a huge help to the ComponentizeJS work. If we need to figure out a name for such namespacing, perhaps we need to do some bikeshedding on that? `smwcg_builtins::` or something?

Reply:
That works for me. We could also move `HostString` somewhere else, as it's just a thin wrapper over spidermonkey types.

Reply:
Do you mind if we leave this to a follow-up PR? My goal with this PR was  to continue breaking up some of the larger headers and reduce dependence on apis that can change when we re-generate the fastly-world c bindings.

Reply:
Certainly, that would be fine.